cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -83,20 +82,30 @@ def process_data(old_data, new_data)
       new_data.reverse_each do |stackname, data|
         reader = data[:loader].reader
         set_env_variables(stackname, current_grid) # set envs for execution time
-        data[:stack_data] = reader.execute(
-          values: data[:variables],
-          defaults: old_data[stackname].nil? ? nil : old_data[stackname][:stack_data]['variables'],
+        values = data[:variables]
+        if old_stack = old_data[stackname]
+          values = values.merge(old_stack[:stack_data]['variables'])
+        end
+        parsed_stack = reader.execute(
+          values: values,
```

Comment:
What happened to `values` vs `defaults`? Without the use of `defaults`, `kontena stack upgrade` is no longer prompting for variables at all, since it gets all of them directly in `values`.... that's only meant for overrides like `-v foo=bar`.

Reply:
This was just an experiment (probably topic of another issue/pr)... for usability it would probably make sense to prompt only new variables by default (with option to force prompt everything).